### ISSUE TITLE
google_checks.xml: LocalVariableName/LocalFinalVariableName false positive for identifier example "guava33_4_6" from Google Java Style Guide

### ISSUE THREAD

**Issue by PaulLieverse-TomTom** [checkstyle/checkstyle#21101]
I have read check documentation: https://checkstyle.org/checks/naming/localfinalvariablename.html and https://checkstyle.org/checks/naming/localvariablename.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat Test.java
// Examples from https://google.github.io/styleguide/javaguide.html#s5.3-camel-case
class Test {
  static void methodLocalVariableName() {
    int newCustomerId = 1;
    Object innerStopwatch = null;
    boolean supportsIpv6OnIos = true;
    boolean turnOn2sv = false;
    int guava33_4_6 = 23; // This reports failure from 6.x onwards
  }

  static void methodLocalFinalVariableName() {
    final int newCustomerId = 1;
    final Object innerStopwatch = null;
    final boolean supportsIpv6OnIos = true;
    final boolean turnOn2sv = false;
    final int guava33_4_6 = 23; // This reports failure starting at 13.9.0
  }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-13.9.0-all.jar -c /google_checks.xml Test.java
Starting audit...
[WARN] /var/tmp/Test.java:8:9: Local variable name 'guava33_4_6' must match pattern '^(_|[a-z]([a-z0-9][a-zA-Z0-9]*)?)$'. [LocalVariableName]
[WARN] /var/tmp/Test.java:16:15: Local final variable name 'guava33_4_6' must match pattern '^(_|[a-z]([a-z0-9][a-zA-Z0-9]*)?)$'. [LocalFinalVariableName]
Audit done.

/var/tmp $ java $RUN_LOCALE -jar checkstyle-13.8.0-all.jar -c /google_checks.xml Test.java
Starting audit...
[WARN] /var/tmp/Test.java:8:9: Local variable name 'guava33_4_6' must match pattern '^(_|[a-z]([a-z0-9][a-zA-Z0-9]*)?)$'. [LocalVariableName]
Audit done.
```

---

According to the Google Java Style Guide, `guava_33_4_6` is an allowed local variable name; see the table in https://google.github.io/styleguide/javaguide.html#s5.3-camel-case, referred to from the guideline for local variable names: https://google.github.io/styleguide/javaguide.html#s5.2.7-local-variable-names
Quoting from that first link:
> In very rare circumstances (for example, multipart version numbers), you may need to use underscores to separate adjacent numbers, since numbers do not have upper and lower case variants. 

However, since CheckStyle 13.9.0, I guess triggered by https://github.com/checkstyle/checkstyle/issues/20562, that local variable name is no longer accepted when using `google_checks.xml` for final local variables. For non-final local variables, that false positive actually has been present since at least CheckStyle 7.0.

**Comment by romani** [comment]
@Atharv3221 , please address this issue

**Comment by romani** [comment]
Fix is merged
